ddtrace/tracer/tracer_test.go

```diff
@@ -2199,11 +2198,18 @@ func BenchmarkStartSpanConcurrent(b *testing.B) {
 			for n := 0; n < b.N; n++ {
 				s, ok := SpanFromContext(ctx)
 				if !ok {
-					b.Fatal("no span")
+					<-err
 				}
 				StartSpan("op", ChildOf(s.Context()))
 			}
 		}()
+
+		select {
+		case <-err:
+			b.Fatal("no span")
+		default:
+			continue
+		}
```

Comment:
I think this isn't the right place for this check. The goroutines above won't actually get to the main loop until `start` is closed below, so there won't ever be anything in `err` at this point.

I assume the linter was calling out that the `b.Fatal` in the goroutine won't actually stop the test when it's called from a child goroutine, which is true. But I think we could also fix it by changing the `b.Fatal` to something like this:

```
b.Fail("no span")
return
```

So, the test is marked as failed (fine to do from the child goroutine) and the goroutine will exit normally. And then when all the child goroutines are done and the benchmark ends, the failure will be reported. WDYT?

Reply:
Got it, makes sense! Changing it to `b.Error()` to maintain the old "no span" message.